### PR TITLE
Add sanitized builds to Kokoro

### PR DESCRIPTION
**Pull request by mkruskal-google**
This will parametrize the Bazel kokoro presubmit, running all of optimized, debug, asan, msan, tsan, and ubsan.

### CODE REVIEW

File: WORKSPACE [hunk 1]
```diff
@@ -16,6 +16,13 @@ http_archive(
     ],
 )
 
+http_archive(
```
Comment (perezd) [inline WORKSPACE:19]: This should live in protobuf_deps.bzl under a maybe.

File: protobuf_deps.bzl [hunk 2]
```diff
@@ -49,9 +49,9 @@ def protobuf_deps():
         http_archive(
             name = "zlib",
             build_file = "@com_google_protobuf//:third_party/zlib.BUILD",
-            sha256 = "629380c90a77b964d896ed37163f5c3a34f6e6d897311f1df2a7016355c45eff",
-            strip_prefix = "zlib-1.2.11",
-            urls = ["https://github.com/madler/zlib/archive/v1.2.11.tar.gz"],
+            sha256 = "d8688496ea40fb61787500e863cc63c9afcbc524468cedeb478068924eb54932",
```
Comment (perezd) [inline protobuf_deps.bzl:52]: was this change necessary in this PR? this just is a good thing to keep separate typically RE: rollbacks and cherrypicks etc.

**Review by perezd — CHANGES_REQUESTED** [review]

File: protobuf_deps.bzl (hunk 2 above)
Comment (mkruskal-google) [reply protobuf_deps.bzl:52]: Yep, this was necessary :). There was a UBSAN bug in zlib that was fixed with 1.2.12

File: WORKSPACE (hunk 1 above)
Comment (mkruskal-google) [reply WORKSPACE:19]: I don't think it should.  This dependency is brought in by googletest, which we don't propagate to clients via protobuf_deps.  Note: *this* one is somewhat unrelated, I hit errors that were likely pre-existing due to this missing dependency.  I suspect the root cause is from when we upgraded googletest

**Review by perezd — APPROVED** [review]
LGTM
